### PR TITLE
Docker local dev

### PR DESCRIPTION
**Pull request by bnlb**
This PR significantly improves the developer experience in this repo by using `docker-compose` to abstract away a lot of pain points.

💡 One note is that there are some assets (`/v1/config/logo`) that are served by `web-server` upstream from `dev-center` and 404 when working locally. This is expected and doesn't represent an issue. You can see this working fine on this cluster: https://nblack.algo.click/developers. I opted not to fix for now since it doesn't seem worth the time, but can revisit in the future if anyone prefers.

Production build deployed here 👉 https://nblack.algo.click/developers

## Local Dev Steps

1.  Check out this branch
2. `yarn setup`
3. `yarn dev`

## Fresh repo initialization with `yarn setup`

```zsh
❯ git clone https://github.com/algorithmiaio/dev-center.git dc
Cloning into 'dc'...
remote: Enumerating objects: 298, done.
remote: Counting objects: 100% (298/298), done.
remote: Compressing objects: 100% (231/231), done.
remote: Total 24581 (delta 152), reused 153 (delta 63), pack-reused 24283
Receiving objects: 100% (24581/24581), 162.71 MiB | 10.90 MiB/s, done.
Resolving deltas: 100% (14133/14133), done.

~/work 17s
❯ ./dc

~/work/dc develop
❯ gco docker-local-dev
Branch 'docker-local-dev' set up to track remote branch 'docker-local-dev' from 'origin'.
Switched to a new branch 'docker-local-dev'

~/work/dc docker-local-dev
❯ yarn setup
yarn run v1.22.10
$ rm -rf vendor && git submodule init && git submodule update && docker run -v $PWD:/jekyll algorithmiahq/dev-center:local-dev-jekyll-server bundle install
Submodule 'api-docs' (https://github.com/algorithmiaio/api-docs.git) registered for path 'api-docs'
Cloning into '/Users/nlb/work/dc/api-docs'...
Submodule path 'api-docs': checked out '2187b58933c65805a79524806a76d3f0ff290f21'
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.
Fetching gem metadata from https://rubygems.org/.........
Fetching rake 13.0.3
Installing rake 13.0.3
Using bundler 2.2.14
Fetching unf_ext 0.0.7.7
Fetching public_suffix 4.0.6
Fetching concurrent-ruby 1.1.8
Fetching colorator 1.1.0
Fetching http_parser.rb 0.6.0
Fetching eventmachine 1.2.7
Installing colorator 1.1.0
Installing public_suffix 4.0.6
Installing eventmachine 1.2.7 with native extensions
Fetching multipart-post 2.1.1
Installing http_parser.rb 0.6.0 with native extensions
Installing concurrent-ruby 1.1.8
Installing multipart-post 2.1.1
Installing unf_ext 0.0.7.7 with native extensions
Fetching ruby2_keywords 0.0.4
Installing ruby2_keywords 0.0.4
Fetching fastimage 2.1.7
Installing fastimage 2.1.7
Fetching forwardable-extended 2.6.0
Fetching ffi 1.14.2
Installing forwardable-extended 2.6.0
Installing ffi 1.14.2 with native extensions
Fetching gemoji 3.0.1
Installing gemoji 3.0.1
Fetching rb-fsevent 0.10.4
Fetching liquid 4.0.3
Fetching rchardet 1.8.0
Fetching rexml 3.2.4
Installing rb-fsevent 0.10.4
Installing liquid 4.0.3
Installing rchardet 1.8.0
Installing rexml 3.2.4
Fetching mercenary 0.4.0
Fetching safe_yaml 1.0.5
Fetching unicode-display_width 1.7.0
Fetching rouge 3.26.0
Fetching posix-spawn 0.3.15
Fetching libv8 3.16.14.19 (x86_64-linux)
Installing safe_yaml 1.0.5
Installing unicode-display_width 1.7.0
Installing mercenary 0.4.0
Installing posix-spawn 0.3.15 with native extensions
Installing rouge 3.26.0
Fetching nesty 1.0.2
Installing nesty 1.0.2
Installing libv8 3.16.14.19 (x86_64-linux)
Fetching redcarpet 3.5.1
Fetching ref 2.0.0
Fetching racc 1.5.2
Fetching faraday 1.1.0
Fetching addressable 2.7.0
Installing redcarpet 3.5.1 with native extensions
Installing ref 2.0.0
Installing addressable 2.7.0
Installing faraday 1.1.0
Installing racc 1.5.2 with native extensions
Fetching i18n 1.8.9
Installing i18n 1.8.9
Fetching unf 0.1.4
Installing unf 0.1.4
Fetching git 1.8.1
Fetching kramdown 2.3.1
Fetching pathutil 0.16.2
Installing pathutil 0.16.2
Installing git 1.8.1
Installing kramdown 2.3.1
Fetching sassc 2.4.0
Fetching terminal-table 2.0.0
Fetching rb-inotify 0.10.1
Fetching em-websocket 0.5.2
Fetching therubyracer 0.12.3
Installing rb-inotify 0.10.1
Installing em-websocket 0.5.2
Installing terminal-table 2.0.0
Installing therubyracer 0.12.3 with native extensions
Installing sassc 2.4.0 with native extensions
Fetching domain_name 0.5.20190701
Installing domain_name 0.5.20190701
Fetching faraday-encoding 0.0.5
Installing faraday-encoding 0.0.5
Fetching faraday-http-cache 2.2.0
Installing faraday-http-cache 2.2.0
Fetching faraday_middleware 1.0.0
Installing faraday_middleware 1.0.0
Fetching nokogiri 1.11.1 (x86_64-linux)
Fetching listen 3.4.1
Installing listen 3.4.1
Fetching http-cookie 1.0.3
Installing http-cookie 1.0.3
Installing nokogiri 1.11.1 (x86_64-linux)
Fetching faraday-cookie_jar 0.0.7
Fetching jekyll-watch 2.2.1
Installing faraday-cookie_jar 0.0.7
Installing jekyll-watch 2.2.1
Fetching kramdown-parser-gfm 1.1.0
Fetching metainspector 5.11.1
Fetching jekyll-sass-converter 2.1.0
Installing kramdown-parser-gfm 1.1.0
Installing jekyll-sass-converter 2.1.0
Installing metainspector 5.11.1
Fetching jekyll 4.2.0
Installing jekyll 4.2.0
Fetching jekyll-redirect-from 0.16.0
Fetching jekyll-last-modified-at 1.3.0
Fetching jekyll-sitemap 1.4.0
Installing jekyll-last-modified-at 1.3.0
Installing jekyll-redirect-from 0.16.0
Installing jekyll-sitemap 1.4.0
Bundle complete! 11 Gemfile dependencies, 56 gems now installed.
Bundled gems are installed into `./vendor/bundle`
✨  Done in 314.56s.
```

## Normal development workflow with `yarn dev`

```zsh
❯ yarn dev
yarn run v1.22.10
$ docker-compose up
Creating network "dc_default" with the default driver
Creating dc_jekyll_1 ... done
Creating dc_slate_1  ... done
Creating dc_node_1   ... done
Attaching to dc_jekyll_1, dc_slate_1, dc_node_1
node_1    | {"name":"dev-center-server","hostname":"e395c9698b62","pid":1,"level":30,"msg":"Starting server","time":"2021-04-06T16:44:43.634Z","v":0}
node_1    | [HPM] Proxy created: /  -> http://slate:4567
node_1    | [HPM] Proxy rewrite rule created: "^/developers/api/" ~> ""
node_1    | [HPM] Proxy created: /  -> http://jekyll:4001
node_1    | {"name":"dev-center-server","hostname":"e395c9698b62","pid":1,"level":30,"msg":"Server started on port 4000.","time":"2021-04-06T16:44:43.747Z","v":0}
slate_1   | == The Middleman is loading
slate_1   | == Sprockets will render css with ruby sass
slate_1   |    consider using Sprockets 4.x to render with SassC
slate_1   | == View your site at "http://localhost:4567", "http://127.0.0.1:4567"
slate_1   | == Inspect your site configuration at "http://localhost:4567/__middleman", "http://127.0.0.1:4567/__middleman"
jekyll_1  | Configuration file: /jekyll/_config.yml
jekyll_1  | Configuration file: /jekyll/_config-dev.yml
jekyll_1  |             Source: /jekyll
jekyll_1  |        Destination: /jekyll/_site
jekyll_1  |  Incremental build: disabled. Enable with --incremental
jekyll_1  |       Generating...
jekyll_1  |                     done in 29.591 seconds.
jekyll_1  |  Auto-regeneration: enabled for '/jekyll'
jekyll_1  |     Server address: http://0.0.0.0:4001/developers/
jekyll_1  |   Server running... press ctrl-c to stop.
```

## Dev center + API docs pages running locally
Screen Shot 2021-04-06 at 9 47 38 AM
Screen Shot 2021-04-06 at 9 45 53 AM

### CODE REVIEW

File: server/prometheus.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-const prometheus = require('prom-client')
-const requestStats = require('request-stats')
+const prometheus = require('prom-client');
```
Comment (bnlb) [inline server/prometheus.js:1]: No code changes in this file, just ran Prettier over it.

File: server/index.spec.js [hunk 2]
```diff
@@ -1,249 +1,253 @@
-const { expect } = require('chai')
-require('mocha')
-const axios = require('axios')
-const shell = require('shelljs')
-const os = require('os')
+const { expect } = require('chai');
```
Comment (bnlb) [inline server/index.spec.js:1]: No code changes in this file, just ran Prettier over it.

**Comment by warmbowski** [comment]
Bah, what are those codacy problems. Looks like you fixed a bunch of stuff :)

File: _layouts/default.html [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
     <link rel="stylesheet" href="{{ site.cdnurl }}{{ site.baseurl }}/css/pygments-github.css">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/vendors.css">
-    <link rel="stylesheet" href="{{ site.baseurl }}/synapse/dist/scoped-synapse.css">
+    <link rel="stylesheet" href="{{ site.baseurl }}/css/scoped-synapse.css?v=2">
```
Comment (warmbowski) [inline _layouts/default.html:29]: So if there is any change to synapse css, it looks like we have to copy in the rebuilt css, eh? Should something be added to the Readme about synapse css?

**Review by warmbowski — COMMENTED** [review]
One comment about Synapse.

File: _layouts/default.html (hunk 3 above)
Comment (bnlb) [reply _layouts/default.html:29]: Nice catch, I'll update.

Comment (bnlb) [reply _layouts/default.html:29]: Updated!

**Review by warmbowski — DISMISSED** [review]
🐑 it

**Review by stevielkim — APPROVED** [review]

**Pull request merged**